test/new-e2e/tests/agent-platform/common/agent_behaviour.go

```diff
@@ -0,0 +1,206 @@
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	e2eClient "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e/client"
+	"github.com/stretchr/testify/require"
+)
+
+// CheckAgentBehaviour runs test to check the agent is behaving as expected
+func CheckAgentBehaviour(t *testing.T, client *ExtendedClient) {
+	t.Run("datadog-agent service running", func(tt *testing.T) {
+		_, err := client.SvcManager.Status("datadog-agent")
+		require.NoError(tt, err, "datadog-agent service should be running")
+	})
+
+	t.Run("datadog-agent checks running", func(tt *testing.T) {
+		var statusOutputJSON map[string]any
+		result := false
+		for try := 0; try < 5 && !result; try++ {
+			err := json.Unmarshal([]byte(client.AgentClient.Status(e2eClient.WithArgs([]string{"-j"})).Content), &statusOutputJSON)
+			require.NoError(tt, err)
+			if runnerStats, ok := statusOutputJSON["runnerStats"]; ok {
+				runnerStatsMap := runnerStats.(map[string]any)
+				if checks, ok := runnerStatsMap["Checks"]; ok {
+					checksMap := checks.(map[string]any)
+					result = len(checksMap) > 0
+				}
+			}
+			time.Sleep(1 * time.Second)
+		}
+		require.True(tt, result, "status output should contain running checks")
+	})
+
+	t.Run("status command infos", func(tt *testing.T) {
+		statusOutput := client.AgentClient.Status().Content
+		require.Contains(tt, statusOutput, "Forwarder")
+		require.Contains(tt, statusOutput, "Host Info")
+		require.Contains(tt, statusOutput, "DogStatsD")
+	})
+
+	t.Run("status command no errors", func(tt *testing.T) {
+		statusOutput := client.AgentClient.Status().Content
+
+		// API Key is invalid we should not check for the following error
+		statusOutput = strings.ReplaceAll(statusOutput, "[ERROR] API Key is invalid", "API Key is invalid")
+		require.NotContains(tt, statusOutput, "ERROR")
+	})
+}
+
+// CheckAgentStops runs tests to check the agent can stop properly
+func CheckAgentStops(t *testing.T, client *ExtendedClient) {
+	t.Run("stops", func(tt *testing.T) {
+		_, err := client.SvcManager.Stop("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Status("datadog-agent")
+		require.Error(tt, err, "datadog-agent service should be stopped")
+	})
+
+	t.Run("refuse connections", func(tt *testing.T) {
+		_, err := client.AgentClient.StatusWithError()
+		require.Error(tt, err, "status should error")
+	})
+
+	t.Run("no running processes", func(tt *testing.T) {
+		agentProcesses := []string{"datadog-agent", "system-probe", "security-agent"}
+		for _, process := range agentProcesses {
+			_, err := client.VMClient.ExecuteWithError(fmt.Sprintf("pgrep -f %s", process))
+			require.Error(tt, err, fmt.Sprintf("process %s should not be running", process))
+		}
+	})
+
+	t.Run("starts after stop", func(tt *testing.T) {
+		_, err := client.SvcManager.Start("datadog-agent")
+		require.NoError(tt, err)
+
+		_, err = client.SvcManager.Status("datadog-agent")
+		require.NoError(tt, err, "datadog-agent should be running")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Should this be in a retry ? is `SvcManager` blocking on start ?

Reply:
It is supposed to be blocking, I will add a retry if we realize afterwards that is not working as expected